### DISCUSSION TITLE
`om show`: are nixosConfigurations supported?

### DISCUSSION THREAD

**Discussion by dezren39** [juspay/omnix#231] [Q&A] [answered]
```
osh-0.22.0$ om show .
🐚 nix --extra-experimental-features 'nix-command flakes' show-config --json️
🐚 /nix/store/7cmigngrfvri4f5k0np33dl8mzsadb82-nix-static-x86_64-unknown-linux-musl-2.24.0pre20240801_af10904/bin/nix flake show --legacy --allow-import-from-derivation --json --default-flake-schemas /nix/store/xzalq6mcw0ahyaccab6k98dbx3ll53y6-source .️
Error: Unable to fetch flake

Caused by:
    0: Command error: Process exited unsuccessfully. exit_code=None stderr=evaluating 'nixosConfigurations'...
       evaluating 'nixosConfigurations'...
       evaluating 'nixosConfigurations.nixos'...
       warning: In a derivation named 'neovim-unwrapped-0.10.1', 'structuredAttrs' disables the effect of the derivation attribute 'disallowedRequisites'; use 'outputChecks.<output>.disallowedRequisites' instead
       trace: rust-flake: Using /nix/store/2laagk1kd40rnfk9jpdp15whbbfb2d8b-flakreate/crate.nix
       trace: rust-flake: Using /nix/store/7zckygczcw7rcvygjrhagg8jkiy0gpvv-nix_health/crate.nix
       trace: rust-flake: Using /nix/store/6nyn5bia9hpz8y9sj9da0g6qp3ci7g29-nix_rs/crate.nix
       trace: rust-flake: Using /nix/store/fq3d54yl2pfz5dvzn442h947r9l6rg0i-nixci/crate.nix
       trace: rust-flake: Using /nix/store/yhajwg1sjf6bvlxfcscxwm9sbzq9g3xc-omnix-cli/crate.nix
       trace: rust-flake: Using /nix/store/lm9bqn4jhi7bfxix9mpqkzvqgd7zccr0-omnix-gui/crate.nix

    1: Process exited unsuccessfully. exit_code=None stderr=evaluating 'nixosConfigurations'...
       evaluating 'nixosConfigurations'...
       evaluating 'nixosConfigurations.nixos'...
       warning: In a derivation named 'neovim-unwrapped-0.10.1', 'structuredAttrs' disables the effect of the derivation attribute 'disallowedRequisites'; use 'outputChecks.<output>.disallowedRequisites' instead
       trace: rust-flake: Using /nix/store/2laagk1kd40rnfk9jpdp15whbbfb2d8b-flakreate/crate.nix
       trace: rust-flake: Using /nix/store/7zckygczcw7rcvygjrhagg8jkiy0gpvv-nix_health/crate.nix
       trace: rust-flake: Using /nix/store/6nyn5bia9hpz8y9sj9da0g6qp3ci7g29-nix_rs/crate.nix
       trace: rust-flake: Using /nix/store/fq3d54yl2pfz5dvzn442h947r9l6rg0i-nixci/crate.nix
       trace: rust-flake: Using /nix/store/yhajwg1sjf6bvlxfcscxwm9sbzq9g3xc-omnix-cli/crate.nix
       trace: rust-flake: Using /nix/store/lm9bqn4jhi7bfxix9mpqkzvqgd7zccr0-omnix-gui/crate.nix
```

**Comment by shivaraj-bh** [comment]
Yes, they are supported. What is the host machine you are running `om show` on?

**Reply by dezren39** [reply]
i'm running it on a nixos machine with 'nixos' as the name. it is the same machine that uses the nixosConfiguration which errors above. the configuration does successfully rebuild. i can run whatever commands may help with troubleshooting and report back. i can also link the flake but it is a bit convoluted

**Reply by srid** [reply]
Does your config repo build with `om ci`?

**Reply by shivaraj-bh** [reply]
> i'm running it on a nixos machine with 'nixos' as the name. it is the same machine that uses the nixosConfiguration which errors above. the configuration does successfully rebuild. i can run whatever commands may help with troubleshooting and report back. i can also link the flake but it is a bit convoluted

A link to your flake would be helpful to reproduce.

**Reply by dezren39** [reply]
> Does your config repo build with `om ci`?
It didn't, my overlays attr wasn't appropriately formed as expected.

It worked for `nixos-rebuild` but not when evaluating the full flake with `om ci`.

I fixed this, and now `om ci` builds.

`om show .` still shows `Error: Unable to fetch flake`.

One thought that that occurred to me was that this flake does have a local subflake referenced with `path:./pkgs/vim`.

Here is a link, but the flake is a bit of a mess-- so sorry about that:
- https://github.com/developing-today/code/blob/0eb78e2ddbbcbb2dd5966ed20a956ccd88ebf101/flake.nix

I'm in the middle of a few different things right now and haven't done any work to make things pretty so there's a bunch of code all over. Again sorry, this is not a minimal reproduction in any way just my current flake as it is right now. It's a bit of a junk drawer of things I'm testing.

**Reply by shivaraj-bh** [reply, marked as answer]
The schema used with `nix flake show` tries to force evaluate the `config.system.build.toplevel` to fetch the `name` and `description` of the derivation (See https://github.com/DeterminateSystems/flake-schemas/blob/0a5c42297d870156d9c57d8f99e476b738dcd982/flake.nix#L254). 

While evaluating this derivation, it fails because of using relative path in your flake input. https://github.com/NixOS/nix/issues/3978#issuecomment-952418478 might answer why this isn’t possible yet. 


For now, either the flake input shouldn’t be a relative path or you could configure a custom schema that doesn’t evaluate `config.system.build.toplevel` for `nixosConfigurations`. The latter is still not very straightforward; this is actually a good idea for a future version of Omnix

**Reply by srid** [reply]
Or use `--override-input` (à la nixos-buffet),
https://github.com/srid/nixos-flake/issues/51#issuecomment-2156063476

Which can even be supported in omnix via `overrideInputs` CI config? i.e, `om run ...`

**Reply by dezren39** [reply]
thank you both! that relative nix flake has been a source of nothing but trouble for me. :-)
